diff -ur chromium-107.0.5304.87.orig/media/gpu/v4l2/v4l2_video_decoder_delegate_vp9.cc chromium-107.0.5304.87/media/gpu/v4l2/v4l2_video_decoder_delegate_vp9.cc
--- chromium-107.0.5304.87.orig/media/gpu/v4l2/v4l2_video_decoder_delegate_vp9.cc	2022-10-27 19:32:44.090105000 -0400
+++ chromium-107.0.5304.87/media/gpu/v4l2/v4l2_video_decoder_delegate_vp9.cc	2022-12-10 16:05:54.899658275 -0500
@@ -273,11 +273,11 @@
                                 &v4l2_frame_params.quant);
   FillV4L2VP9SegmentationParams(segm_params, &v4l2_frame_params.seg);
 
-  std::vector<struct v4l2_ext_control> ext_ctrls = {
+  const struct v4l2_ext_control frame_ctrl =
       {.id = V4L2_CID_STATELESS_VP9_FRAME,
        .size = sizeof(v4l2_frame_params),
-       .ptr = &v4l2_frame_params},
-  };
+       .ptr = &v4l2_frame_params};
+  std::vector<struct v4l2_ext_control> ext_ctrls = { frame_ctrl };
 
   struct v4l2_ctrl_vp9_compressed_hdr v4l2_compressed_hdr_probs;
   if (supports_compressed_header_) {
@@ -285,9 +285,11 @@
     v4l2_compressed_hdr_probs.tx_mode = frame_hdr->compressed_header.tx_mode;
     FillV4L2VP9ProbsParams(frame_hdr->frame_context,
                            &v4l2_compressed_hdr_probs);
-    ext_ctrls.push_back({.id = V4L2_CID_STATELESS_VP9_COMPRESSED_HDR,
-                         .size = sizeof(v4l2_compressed_hdr_probs),
-                         .ptr = &v4l2_compressed_hdr_probs});
+    const struct v4l2_ext_control compressed_hdr_ctrl =
+        {.id = V4L2_CID_STATELESS_VP9_COMPRESSED_HDR,
+         .size = sizeof(v4l2_compressed_hdr_probs),
+         .ptr = &v4l2_compressed_hdr_probs};
+    ext_ctrls.push_back(compressed_hdr_ctrl);
   }
 
   const __u32 ext_ctrls_size = base::checked_cast<__u32>(ext_ctrls.size());
