diff --git a/ui/gfx/font_render_params_linux.cc b/ui/gfx/font_render_params_linux.cc
index 3ab8b2ed3e..eed71df9c6 100644
--- a/ui/gfx/font_render_params_linux.cc
+++ b/ui/gfx/font_render_params_linux.cc
@@ -226,21 +226,6 @@ FontRenderParams GetFontRenderParams(const FontRenderParamsQuery& query,
     params.hinting = FontRenderParams::HINTING_FULL;
     params.subpixel_rendering = FontRenderParams::SUBPIXEL_RENDERING_NONE;
     params.subpixel_positioning = false;
-  } else if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
-                 switches::kDisableFontSubpixelPositioning)) {
-#if !BUILDFLAG(IS_CHROMEOS_ASH)
-    params.subpixel_positioning = actual_query.device_scale_factor > 1.0f;
-#else
-    // We want to enable subpixel positioning for fractional dsf.
-    params.subpixel_positioning =
-        std::abs(std::round(actual_query.device_scale_factor) -
-                 actual_query.device_scale_factor) >
-        std::numeric_limits<float>::epsilon();
-#endif  // !BUILDFLAG(IS_CHROMEOS_ASH)
-
-    // To enable subpixel positioning, we need to disable hinting.
-    if (params.subpixel_positioning)
-      params.hinting = FontRenderParams::HINTING_NONE;
   }
 
   // Use the first family from the list if Fontconfig didn't suggest a family.
