diff -ur chromium-107.0.5304.87.orig/media/gpu/v4l2/BUILD.gn chromium-107.0.5304.87/media/gpu/v4l2/BUILD.gn
--- chromium-107.0.5304.87.orig/media/gpu/v4l2/BUILD.gn	2022-10-27 19:32:44.078105000 -0400
+++ chromium-107.0.5304.87/media/gpu/v4l2/BUILD.gn	2022-12-09 16:31:37.867627994 -0500
@@ -56,8 +56,6 @@
     "v4l2_video_decoder_backend_stateful.h",
     "v4l2_video_decoder_backend_stateless.cc",
     "v4l2_video_decoder_backend_stateless.h",
-    "v4l2_video_decoder_delegate_av1.cc",
-    "v4l2_video_decoder_delegate_av1.h",
     "v4l2_video_decoder_delegate_h264.cc",
     "v4l2_video_decoder_delegate_h264.h",
     "v4l2_video_decoder_delegate_vp8.cc",
