diff --git a/xwayland/selection/incoming.c b/xwayland/selection/incoming.c
index 791eb5cf..43aeac54 100644
--- a/xwayland/selection/incoming.c
+++ b/xwayland/selection/incoming.c
@@ -467,13 +467,6 @@ void xwm_handle_selection_notify(struct wlr_xwm *xwm,
 			xwm_selection_transfer_destroy(transfer);
 		}
 	} else if (event->target == xwm->atoms[TARGETS]) {
-		// No xwayland surface focused, deny access to clipboard
-		if (xwm->focus_surface == NULL) {
-			wlr_log(WLR_DEBUG, "denying write access to clipboard: "
-				"no xwayland surface focused");
-			return;
-		}
-
 		// This sets the Wayland clipboard (by calling wlr_seat_set_selection)
 		xwm_selection_get_targets(selection);
 	} else if (transfer) {
diff --git a/xwayland/selection/outgoing.c b/xwayland/selection/outgoing.c
index 87c0fb06..59b233da 100644
--- a/xwayland/selection/outgoing.c
+++ b/xwayland/selection/outgoing.c
@@ -410,15 +410,6 @@ void xwm_handle_selection_request(struct wlr_xwm *xwm,
 		return;
 	}
 
-	// No xwayland surface focused, deny access to clipboard
-	if (xwm->focus_surface == NULL && xwm->drag_focus == NULL) {
-		char *selection_name = xwm_get_atom_name(xwm, selection->atom);
-		wlr_log(WLR_DEBUG, "denying read access to selection %u (%s): "
-			"no xwayland surface focused", selection->atom, selection_name);
-		free(selection_name);
-		goto fail_notify_requestor;
-	}
-
 	if (req->target == xwm->atoms[TARGETS]) {
 		xwm_selection_send_targets(selection, req);
 	} else if (req->target == xwm->atoms[TIMESTAMP]) {
