diff -ur wlroots-0.17.0.orig/xwayland/selection/incoming.c wlroots-0.17.0/xwayland/selection/incoming.c
--- wlroots-0.17.0.orig/xwayland/selection/incoming.c	2023-11-21 11:06:13.000000000 -0500
+++ wlroots-0.17.0/xwayland/selection/incoming.c	2023-12-21 17:21:28.300738505 -0500
@@ -464,13 +464,6 @@
 			xwm_selection_transfer_destroy(transfer);
 		}
 	} else if (event->target == xwm->atoms[TARGETS]) {
-		// No xwayland surface focused, deny access to clipboard
-		if (xwm->focus_surface == NULL) {
-			wlr_log(WLR_DEBUG, "denying write access to clipboard: "
-				"no xwayland surface focused");
-			return;
-		}
-
 		// This sets the Wayland clipboard (by calling wlr_seat_set_selection)
 		xwm_selection_get_targets(selection);
 	} else if (transfer) {
diff -ur wlroots-0.17.0.orig/xwayland/selection/outgoing.c wlroots-0.17.0/xwayland/selection/outgoing.c
--- wlroots-0.17.0.orig/xwayland/selection/outgoing.c	2023-11-21 11:06:13.000000000 -0500
+++ wlroots-0.17.0/xwayland/selection/outgoing.c	2023-12-21 17:21:38.167868245 -0500
@@ -409,17 +409,6 @@
 		return;
 	}
 
-	// No xwayland surface focused, deny access to clipboard
-	if (xwm->focus_surface == NULL && xwm->drag_focus == NULL) {
-		if (wlr_log_get_verbosity() >= WLR_DEBUG) {
-			char *selection_name = xwm_get_atom_name(xwm, selection->atom);
-			wlr_log(WLR_DEBUG, "denying read access to selection %u (%s): "
-				"no xwayland surface focused", selection->atom, selection_name);
-			free(selection_name);
-		}
-		goto fail_notify_requestor;
-	}
-
 	if (req->target == xwm->atoms[TARGETS]) {
 		xwm_selection_send_targets(selection, req);
 	} else if (req->target == xwm->atoms[TIMESTAMP]) {
