diff -ur go-md2man-2.0.3.orig/Makefile go-md2man-2.0.3/Makefile
--- go-md2man-2.0.3.orig/Makefile	2024-12-18 20:48:38.783538660 -0500
+++ go-md2man-2.0.3/Makefile	2024-12-18 20:48:48.035601886 -0500
@@ -16,7 +16,7 @@
 
 bin/go-md2man: actual_build_flags := $(BUILD_FLAGS) -o bin/go-md2man
 bin/go-md2man: bin
-	@CGO_ENABLED=0 go build $(actual_build_flags)
+	@CGO_ENABLED=1 go build $(actual_build_flags)
 
 bin:
 	@mkdir ./bin
