diff --git a/startkde/plasma-session/startup.cpp b/startkde/plasma-session/startup.cpp
index 727c040..55f5015 100644
--- a/startkde/plasma-session/startup.cpp
+++ b/startkde/plasma-session/startup.cpp
@@ -160,19 +160,6 @@ Startup::Startup(QObject *parent)
         } else {
             x11WindowManagerJob = new StartServiceJob(windowManager, {}, {});
         }
-    } else {
-        // This must block until started as it sets the WAYLAND_DISPLAY/DISPLAY env variables needed for the rest of the boot
-        // fortunately it's very fast as it's just starting a wrapper
-        StartServiceJob kwinWaylandJob(QStringLiteral("kwin_wayland_wrapper"), {QStringLiteral("--xwayland")}, QStringLiteral("org.kde.KWinWrapper"));
-        kwinWaylandJob.exec();
-        // kslpash is only launched in plasma-session from the wayland mode, for X it's in startplasma-x11
-
-        const KConfig cfg(QStringLiteral("ksplashrc"));
-        // the splashscreen and progress indicator
-        KConfigGroup ksplashCfg = cfg.group("KSplash");
-        if (ksplashCfg.readEntry("Engine", QStringLiteral("KSplashQML")) == QLatin1String("KSplashQML")) {
-            QProcess::startDetached(QStringLiteral("ksplashqml"), {});
-        }
     }
 
     // Keep for KF5; remove in KF6 (KInit will be gone then)
