diff --git a/startkde/startplasma-wayland.cpp b/startkde/startplasma-wayland.cpp
index b369ef9..a3ea4fa 100644
--- a/startkde/startplasma-wayland.cpp
+++ b/startkde/startplasma-wayland.cpp
@@ -28,7 +28,6 @@ int main(int argc, char **argv)
     QCoreApplication app(argc, argv);
 
     createConfigDirectory();
-    setupCursor(true);
 
     {
         KConfig fonts(QStringLiteral("kcmfonts"));
diff --git a/startkde/startplasma-x11.cpp b/startkde/startplasma-x11.cpp
index d1a1a58..0a9b43f 100644
--- a/startkde/startplasma-x11.cpp
+++ b/startkde/startplasma-x11.cpp
@@ -84,7 +84,6 @@ int main(int argc, char **argv)
         }
     }
 
-    setupCursor(false);
     setupFontDpi();
     QScopedPointer<QProcess, KillBeforeDeleter> ksplash(setupKSplash());
 
