diff -ur firefox-122.0.1.orig/dom/media/flac/FlacDecoder.cpp firefox-122.0.1/dom/media/flac/FlacDecoder.cpp
--- firefox-122.0.1.orig/dom/media/flac/FlacDecoder.cpp	2024-03-09 22:30:55.128415018 -0500
+++ firefox-122.0.1/dom/media/flac/FlacDecoder.cpp	2024-03-09 22:30:58.533421837 -0500
@@ -17,7 +17,7 @@
 #elif defined(MOZ_FFMPEG)
   RefPtr<PDMFactory> platform = new PDMFactory();
   return StaticPrefs::media_flac_enabled() &&
-         (platform->SupportsMimeType("audio/flac"_ns) != media::DecodeSupport::Unsupported);
+         !platform->SupportsMimeType("audio/flac"_ns).isEmpty();
 #else
   return false;
 #endif
