--- firefox-130.0.1.orig/media/libyuv/libyuv/libyuv.gyp	2024-10-14 15:24:14.110799299 -0400
+++ firefox-130.0.1/media/libyuv/libyuv/libyuv.gyp	2024-10-15 16:51:52.783552810 -0400
@@ -132,6 +132,8 @@
         }],
       ], #conditions
       'defines': [
+        # Firefox build system not updated for this yet
+        'LIBYUV_DISABLE_RVV',
         # Enable the following 3 macros to turn off assembly for specified CPU.
         # 'LIBYUV_DISABLE_X86',
         # 'LIBYUV_DISABLE_NEON',
