diff --git a/contrib/dracut/90zfs/zfs-load-key.sh.in b/contrib/dracut/90zfs/zfs-load-key.sh.in
index c974b3d9e..471af705c 100755
--- a/contrib/dracut/90zfs/zfs-load-key.sh.in
+++ b/contrib/dracut/90zfs/zfs-load-key.sh.in
@@ -45,7 +45,7 @@ if [ "$(zpool list -H -o feature@encryption "${BOOTFS%%/*}")" = 'active' ]; then
         case "${KEYLOCATION%%://*}" in
             prompt)
                 for _ in 1 2 3; do
-                    systemd-ask-password --no-tty "Encrypted ZFS password for ${BOOTFS}" | zfs load-key "${ENCRYPTIONROOT}" && break
+                    systemd-ask-password --no-tty --timeout=0 "Enter passphrase for ${ENCRYPTIONROOT}:" | zfs load-key "${ENCRYPTIONROOT}" && break
                 done
                 ;;
             http*)
