diff --git a/lib/puppet/provider/service/init.rb b/lib/puppet/provider/service/init.rb
index a28c4b8e52..308b4c08b0 100644
--- a/lib/puppet/provider/service/init.rb
+++ b/lib/puppet/provider/service/init.rb
@@ -24,7 +24,7 @@ Puppet::Type.type(:service).provide :init, :parent => :base do
   confine :true => begin
       os = Puppet.runtime[:facter].value('os.name').downcase
       family = Puppet.runtime[:facter].value('os.family').downcase
-      !(os == 'debian' || os == 'ubuntu' || family == 'redhat')
+      !(os == 'debian' || os == 'ubuntu' || family == 'redhat' || family == 'gentoo')
   end
 
   # We can't confine this here, because the init path can be overridden.
