diff --git a/lib/puppet/provider/service/init.rb b/lib/puppet/provider/service/init.rb
index 88e7242545..b86fa5448e 100644
--- a/lib/puppet/provider/service/init.rb
+++ b/lib/puppet/provider/service/init.rb
@@ -22,8 +22,8 @@ Puppet::Type.type(:service).provide :init, :parent => :base do
 
   # Debian and Ubuntu should use the Debian provider.
   confine :false => ['Debian', 'Ubuntu'].include?(Puppet.runtime[:facter].value('os.name'))
-  # RedHat systems should use the RedHat provider.
-  confine :false => Puppet.runtime[:facter].value('os.family') == 'RedHat'
+  # Gentoo and RedHat systems should use the systemd provider.
+  confine :false => ['Gentoo', 'RedHat'].include?(Puppet.runtime[:facter].value('os.family'))
 
   # We can't confine this here, because the init path can be overridden.
   # confine :exists => defpath
diff --git a/lib/puppet/provider/service/systemd.rb b/lib/puppet/provider/service/systemd.rb
index de6acefc92..e415f2259f 100644
--- a/lib/puppet/provider/service/systemd.rb
+++ b/lib/puppet/provider/service/systemd.rb
@@ -13,8 +13,7 @@ Puppet::Type.type(:service).provide :systemd, :parent => :base do
 
   commands :systemctl => "systemctl"
 
-  confine :true => Puppet::FileSystem.exist?('/proc/1/comm') && Puppet::FileSystem.read('/proc/1/comm').include?('systemd')
-
+  defaultfor 'os.family' => :gentoo
   defaultfor 'os.family' => [:archlinux]
   defaultfor 'os.family' => :redhat, 'os.release.major' => ["7", "8", "9"]
   defaultfor 'os.family' => :redhat, 'os.name' => :fedora
