diff --git a/src/corelib/tools/qsimd.cpp b/src/corelib/tools/qsimd.cpp
index 6ad6795f..989ca103 100644
--- a/src/corelib/tools/qsimd.cpp
+++ b/src/corelib/tools/qsimd.cpp
@@ -553,24 +553,6 @@ quint64 qDetectCpuFeatures()
         }
     }
 
-#ifdef RUNNING_ON_VALGRIND
-    bool runningOnValgrind = RUNNING_ON_VALGRIND;
-#else
-    bool runningOnValgrind = false;
-#endif
-    if (Q_UNLIKELY(!runningOnValgrind && minFeature != 0 && (f & minFeature) != minFeature)) {
-        quint64 missing = minFeature & ~f;
-        fprintf(stderr, "Incompatible processor. This Qt build requires the following features:\n   ");
-        for (int i = 0; i < features_count; ++i) {
-            if (missing & (Q_UINT64_C(1) << i))
-                fprintf(stderr, "%s", features_string + features_indices[i]);
-        }
-        fprintf(stderr, "\n");
-        fflush(stderr);
-        qFatal("Aborted. Incompatible processor: missing feature 0x%llx -%s.", missing,
-               features_string + features_indices[qCountTrailingZeroBits(missing)]);
-    }
-
     qt_cpu_features[0].storeRelaxed(f | quint32(QSimdInitialized));
 #ifndef Q_ATOMIC_INT64_IS_SUPPORTED
     qt_cpu_features[1].storeRelaxed(f >> 32);
