--- a/third_party/skia/src/ports/SkFontHost_FreeType.cpp	2017-09-29 13:14:59.607873200 -0400
+++ b/third_party/skia/src/ports/SkFontHost_FreeType.cpp	2017-09-29 13:15:06.452887969 -0400
@@ -677,11 +677,6 @@
         // collapse full->normal hinting if we're not doing LCD
         h = SkPaint::kNormal_Hinting;
     }
-    if ((rec->fFlags & SkScalerContext::kSubpixelPositioning_Flag)) {
-        if (SkPaint::kNo_Hinting != h) {
-            h = SkPaint::kSlight_Hinting;
-        }
-    }
 
     // rotated text looks bad with hinting, so we disable it as needed
     if (!isAxisAligned(*rec)) {
